### PR TITLE
feat: adds ha vpn to the networking blueprint

### PR DESCRIPTION
**Pull request by g-awmalik**
Changes:

- usage of ComputeVPNGateway in vpn.yaml
- exposed peered ip in the top-level setters

### CODE REVIEW

File: catalog/networking/network/subnet/nat.yaml [hunk 1]
```diff
@@ -11,7 +11,6 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
```
Comment (bharathkkb) [inline catalog/networking/network/subnet/nat.yaml:14]: nit: we should keep these newlines

File: catalog/networking/network/setters.yaml [hunk 2]
```diff
@@ -21,6 +21,7 @@ data:
   network-name: network-name
   project-id: project-id
   region: us-central1
+  vpn-tunnel-peerIp: "8.8.8.8"
```
Comment (bharathkkb) [inline catalog/networking/network/setters.yaml:24]:
```suggestion
  vpn-tunnel-peer-ip: "8.8.8.8"
```

File: catalog/networking/network/vpn.yaml [hunk 3]
```diff
@@ -35,17 +48,20 @@ metadata:
...
         name: vpn-shared-secret # kpt-set: ${vpn-secret-name}
         key: vpn-shared-secret # kpt-set: ${vpn-secret-key}
-  targetVPNGatewayRef:
-    name: network-name-vpn-gateway # kpt-set: ${network-name}-vpn-gateway
+  vpnGatewayInterface: 0
+  vpnGatewayRef:
+    name: network-name-ha-vpn-gateway # kpt-set: ${network-name}-ha-vpn-gateway
 ---
 apiVersion: compute.cnrm.cloud.google.com/v1beta1
 kind: ComputeAddress
```
Comment (bharathkkb) [inline catalog/networking/network/vpn.yaml:67]: Do we still need a compute address for HA vpn?

File: catalog/networking/network/vpn.yaml [hunk 4]
```diff
@@ -11,22 +11,35 @@
...
+    cnrm.cloud.google.com/project-id: project-id # kpt-set: ${project-id}
+    cnrm.cloud.google.com/blueprint: cnrm/landing-zone:networking/v0.3.0
+spec:
+  redundancyType: "SINGLE_IP_INTERNALLY_REDUNDANT"
+  interface:
+    - id: 0
+      ipAddress: "8.8.8.8" # kpt-set: ${vpn-tunnel-peerIp}
+---
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
 kind: ComputeVPNTunnel
```
Comment (bharathkkb) [inline catalog/networking/network/vpn.yaml:43]: Shouldn't we be creating two tunnels (assuming https://cloud.google.com/network-connectivity/docs/vpn/concepts/topologies#1-peer-1-address is what you were configuring)

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @g-awmalik 
The [readme](https://github.com/GoogleCloudPlatform/blueprints/tree/main/catalog/networking/network#resources) also needs to be updated with the new resources/setter using [helper](https://github.com/GoogleCloudPlatform/blueprints/blob/main/utils/fix-readmes.sh)

File: catalog/networking/network/vpn.yaml (hunk 3 above)
Comment (g-awmalik) [reply catalog/networking/network/vpn.yaml:67]: We don't, that was a miss from me.

File: catalog/networking/network/subnet/nat.yaml (hunk 1 above)
Comment (g-awmalik) [reply catalog/networking/network/subnet/nat.yaml:14]: Acked.

File: catalog/networking/network/setters.yaml (hunk 2 above)
Comment (g-awmalik) [reply catalog/networking/network/setters.yaml:24]: Acked.

File: catalog/networking/network/vpn.yaml (hunk 4 above)
Comment (g-awmalik) [reply catalog/networking/network/vpn.yaml:43]: Valid point. I've configured the peer gateway to use "TWO_IPS_REDUNDANCY" instead and configured another tunnel so they both have an interface to hook on to.

**Comment by g-awmalik** [comment]
Thanks for the review @bharathkkb. Please review the resolutions one more time.

File: catalog/networking/network/README.md [hunk 5]
```diff
@@ -43,6 +43,8 @@ region                              9
 source-subnetwork-ip-ranges-to-nat  1
 vpn-secret-key                      1
 vpn-secret-name                     1
+vpn-tunnel-peer-ip-01               1
+vpn-tunnel-peer-ip-02               1
```
Comment (bharathkkb) [inline catalog/networking/network/README.md:47]: nit: further down this readme there is a [resources](https://github.com/GoogleCloudPlatform/blueprints/tree/main/catalog/networking/network#resources) section which also needs to be [updated](https://github.com/GoogleCloudPlatform/blueprints/blob/main/utils/fix-readmes.sh)

**Review by bharathkkb — APPROVED** [review]
/cc @morgante is a minor version bump okay since there has not been a 1.0 release?

**Comment by morgante** [comment]
> /cc @morgante is a minor version bump okay since there has not been a 1.0 release?

Yes, minor release is fine.

File: catalog/networking/network/README.md (hunk 5 above)
Comment (g-awmalik) [reply catalog/networking/network/README.md:47]: Acked.

**Pull request merged**